src/query/api/v1/handler/prometheus/remote/attribution.go

```diff
@@ -0,0 +1,84 @@
+package remote
+
+import (
+	"errors"
+	"math/rand"
+	"strings"
+	"sync"
+	"sync/atomic"
+
+	"github.com/m3db/m3/src/query/generated/proto/prompb"
+	"github.com/m3db/m3/src/x/instrument"
+	"github.com/uber-go/tally"
+)
+
+/**
+ * This class attributes metrics with matched configured labels for visibility purpose.
+ * For example, today m3 only emits totally write QPS, with this attribution, we could plot
+ * QPS breakdown by user defined label like namespace in Kubernetes or any label with finite
+ * values (configured via capacity) and business meanings.
+ */
+
+var errNoOption = errors.New("no option configured for promAttributionMetrics")
+
+type promAttributionMetrics struct {
+	opts               *instrument.AttributionConfiguration
+	baseScope          tally.Scope
```

Comment:
Do we need to create new metric name for each attribution? Can we add attribute `label` into a m3 write metrics? In that case we can aggregate the metric based on label.

Reply:
Initially i did use label instead of metric names, but somehow when i add multiple attributions tally will complain with dup metrics even the label values are different.